tools/postgres/diff_schema.js

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+var program = require('commander');
+var randomstring = require("randomstring");
+const execSync = require('child_process').execSync;
+var path = require('path');
+var fs = require('fs');
+var glob = require("glob")
+var tmp = require('tmp');
+
+let POSTGRES_URL = process.env.POSTGRES_URL
+let ORIGINAL_POSTGRES_URL = (typeof POSTGRES_URL !== 'undefined') ? POSTGRES_URL:"jdbc:postgresql://localhost/webknossos"
+const scriptdir = __dirname
+const scriptName = __filename
+
+function dump(parameter) {
+	var cleanUp = function(){};
+	if(parameter == "DB") { 
+		POSTGRES_URL=ORIGINAL_POSTGRES_URL // this environment variable is passed to dump_schema.sh
+	} else {
+		const tempDbName = generateRandomName()
+		const postgresDirname = path.dirname(ORIGINAL_POSTGRES_URL)
+		POSTGRES_URL= postgresDirname + '/' + tempDbName;
+		const dbName = execSync(scriptdir+'/db_name.sh', {env: {'POSTGRES_URL': POSTGRES_URL}}).toString().trim() // "trim" to remove the line break
+		if(dbName !== tempDbName) {
+			console.log("Wrong dbName")	
+			process.exit()
+		}
+		console.log("Creating DB " + dbName)
+		const dbHost = execSync(scriptdir+'/db_host.sh', {env: {'POSTGRES_URL': POSTGRES_URL}}).toString().trim()
+		execSync(`psql -U postgres -h ${dbHost} -c "CREATE DATABASE ${dbName};"`, {env: {'PGPASSWORD': 'postgres'}})
+		var fileNames = glob.sync(parameter)
+		let concatenateFileNames = ""
+		fileNames.forEach(function(fileName) {
+  			concatenateFileNames += " -f " + fileName;
+		});
+		execSync(`psql -U postgres -h ${dbHost} --dbname="${dbName}" -v ON_ERROR_STOP=ON -q ${concatenateFileNames}`, {env: {'PGPASSWORD': 'postgres'}})
+		cleanUp = function(){
```

Comment:
Is there a nicer way to translate `trap` from bash into something in js that is called when the method exits?

Reply:
You don't need an extra function here, I think. The structure could be:
```
execSync(`… CREATE DATABASE …`)
try {
  …
  return …
} catch(err) {
  console.log(err)
} finally {
  console.log(`CLEANUP: DROP DATABASE ${dbName}`)
 	execSync(`psql -U postgres -h ${dbHost} -c \"DROP DATABASE ${dbName};\"`, {env: {'PGPASSWORD': 'postgres'}})
}
```

Is this correct, @normanrz?

Reply:
It does, at least the documentation says so.